src/icon-manifest/iconGenerator.ts

```diff
@@ -0,0 +1,356 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as _ from 'lodash';
+
+import { schema as defaultSchema } from './defaultSchema';
+import { ISettingsManager, SettingsManager } from '../settings';
+import { IExtensionCollection, IFileExtension, IFolderExtension, FileFormat, IExtension } from '../models/IExtension';
+import { IIconSchema } from '../models/IIconSchema';
+import { IVSCode } from '../models/IVSCode';
+import { IIconGenerator } from '../models/IIconGenerator';
+
+// tslint:disable-next-line no-var-requires
+const packageJson = require('../../../package.json');
+
+export class IconGenerator implements IIconGenerator {
+  private settingsManager: ISettingsManager;
+  private iconsFolderPath: string;
+  private manifestFolderPath: string;
+
+  constructor(private vscode: IVSCode) {
+    this.settingsManager = new SettingsManager(vscode);
+    // relative to this file
+    this.iconsFolderPath = path.join(__dirname, '../../../icons');
```

Comment:
It seems to not affect the functionality in Unix but I'll rework it to return `Unix` style paths.